pkg/scheduler/core/generic_scheduler.go

```diff
@@ -404,22 +430,18 @@ func podFitsOnNode(
 	ecache *EquivalenceCache,
 	queue SchedulingQueue,
 	alwaysCheckAllPredicates bool,
+	equivCacheInfo equivalenceClasseInfo,
```

Comment:
I think a pointer to the struct would be better. Then you can pass nil instead of empty struct when no equivalence hash is available. eCacheAvailable should be changed to:
```
eCacheAvailable = equivCacheInfo != nil && equivCacheInfo.isAvailable && !podsAdded
```

Reply:
Aha, I was intentionally use struct to make this `eCacheAvailable`  line here simpler, never mind, let's use pointer.